cf-agent/verify_users_pam.c

```diff
@@ -994,7 +994,8 @@ static bool VerifyIfUserNeedsModifs (const char *puser, User u, const struct pas
             CFUSR_SETBIT (*changemap, i_group);
         }
     }
-    if (u.groups_secondary != NULL)
+
+    if (u.groups_secondary_given)
```

Comment:
@kacf have a quick look in this commit.

As I discussed in the chat, I'm not sure how a users promise should behave if the `groups_secondary` attribute is not present.

In particular it seems weird that having the `groups_secondary` attribute as empty, and not having it at all, alters the behaviour of the promise.

Anyway this commit should only preserve the old behaviour, which was broken after purging `cf_null`.